### ISSUE TITLE
instagram blocks account when using bot 

### ISSUE THREAD

**Issue by Benjuari** [social-manager-tools/socialmanagertools-igbot#196]
when using realistic like-mode with sleep pattern, IG does block account for couple of days.
I tried lowering the likes per day but still results in block.
Maybe the algorithm should start to mimic more organic human use. Maybe too regular mathematical patterns should be avoided. 
Some research is needed to find the cause.

**Comment by ptkdev** [comment]
Remember:
> A bot is a legal software, creating a bot does not violate any law and every day thousands of these software visit sites around the world. Using a bot however violates the terms and conditions of YOUR account on social networks as highlighted in the instagram or facebook policy.

> The developers assume no responsibility for ban, soft ban or permanent ban of YOUR account for violating the terms and conditions of the social network where you intend to use it. This software is provided with no warranty, under the MIT license.

At moment we work at new release and test possible workaround. This happen because 5 days ago instagram updated your website and set new limits. 

The realistic mode is similar at human usage: set fast 10-12 likes, sleep X minutes and set 10-12 likes. Human use instagram in this mode and sleep in night. Imho we need more realistic algorithm but is hard! Stay tuned

**Comment by holdusback** [comment]
I come from https://github.com/social-manager-tools/socialmanagertools-igbot/issues/182

I think I have the issue too, log say it like fotos, but If i go manually in the app on the foto, I see the empty heart...

**Comment by ptkdev** [comment]
@holdusback if you set debug mode to enabled (or `headless` to `true` if you use CLI version) what you see on google chrome?

**Comment by holdusback** [comment]
I use the EXE windows version , enabled the debug mode, now I see the automated chrome window, its look like to really like the post ! Not like before. 

So the debug mode maybe fix my issue ? I will test this full day tomorrow

**Comment by ptkdev** [comment]
@holdusback debug mode disabled run chrome in "hide mode", imho instagram has problem with headless chrome version.

**Comment by holdusback** [comment]
The only difference in the log is that the before & after click tag are in french, before it was in english 
button text before click: <span aria-label="Like"  
to ->
button text before click: <span aria-label="J’aime" 

Maybe he triggered the bot cause it was in english before...

**Comment by lecoueyl** [comment]
If you run without headless option being enable, you can see how the bot behave.
So you can notice that instagram will block the session after a few like using the bot with the following popup:

![action-blocked-tell-us-instagram-screenshot](https://user-images.githubusercontent.com/8327054/66725626-eb1efa80-ee6e-11e9-9613-c3ebb40ffc65.png)

The bot will continue to behave as normal but will not being able to like posts (even if the log show that post are being liked)

**Comment by ptkdev** [comment]
@lecoueyl after august 2019 max action/daily are 300 (not 1000). This is problem.

**Comment by lecoueyl** [comment]
But I think instagram can also detect that the action is performed from a bot because even after 20 likes on the first day, it shows the `Action Blocked` popup.

I think it will be very hard bypass this

**Comment by holdusback** [comment]
There is always a way, maybe a random liker between 0 sec and 60 sec (1 minute), to not like like a robot ... Not a pro but there is always a way in code !

**Comment by ptkdev** [comment]
@lecoueyl 20likes, logout, wait X minutes, login 20likes in loop. Imho that work. We need try this solution :)

**Comment by holdusback** [comment]
@ptkdev Do you know when a new version will be online ?

**Comment by chyan26** [comment]
Another possible solution is to add delay between each like or follow operation.

**Comment by ptkdev** [comment]
@chyan26 nope, now bot work with random delay. 12likes fast, random delay, 12like fast, random delay.

**Comment by holdusback** [comment]
Maybe random like between X and X, random delay would be better. I think with code we can really fake a user

**Comment by chyan26** [comment]
I noticed that IG changed their app.  The message gives you a date of soft ban.  For example, I got 3 days from making likes.

**Comment by SimplePotato** [comment]
What I notice is that I can still like posts on my phone, but when I start the bot on my Pi, I get the ban message. That's weird.

**Comment by SimplePotato** [comment]
Maybe we should try to give the bot to randomly chose the minutes before starting liking again, waiting X seconds before pushing that 💓. Probably the most important human behaviour is to sometimes not to like the post and here and then skip the post without liking it.

**Comment by holdusback** [comment]
@ptkdev Did you find any fix for this ? Im a lil bit scared of using this 0.6.7 version cause of the soft ban I've got one time ahah :p

**Comment by ptkdev** [comment]
@holdusbackno **fix aren't available**, instagram in august 2019 decrease actions/daily to _300_. Set _250_ likes/daily and you not give `change password warning`.
